GVRf/Framework/src/org/gearvrf/GVRViewManager.java

```diff
@@ -246,26 +259,116 @@ void beforeDrawEyes() {
-            if (eye == 1) {
-                mainCameraRig.predict(4.0f / 60.0f);
-                GVRCamera rightCamera = mainCameraRig.getRightCamera();
-                renderCamera(mActivity.appPtr, mMainScene, rightCamera,
-                        mRenderBundle.getRightRenderTexture(), mRenderBundle);
-                mActivity.setCamera(rightCamera);
-            } else {
-                mainCameraRig.predict(3.5f / 60.0f);
-                GVRCamera leftCamera = mainCameraRig.getLeftCamera();
-                renderCamera(mActivity.appPtr, mMainScene, leftCamera,
-                        mRenderBundle.getLeftRenderTexture(), mRenderBundle);
-                mActivity.setCamera(leftCamera);
-            }
-        }
-    }
-
+	void captureScreenLeft(GVRScreenshotCallback callback) {
+		mScreenshotLeftEnabled = true;
+		mScreenshotLeftCallback = callback;
+	}
+
+	void captureScreenRight(GVRScreenshotCallback callback) {
+		mScreenshotRightEnabled = true;
+		mScreenshotRightCallback = callback;
+	}
+
+	void captureScreenCenter(GVRScreenshotCallback callback) {
+		mScreenshotCenterEnabled = true;
+		mScreenshotCenterCallback = callback;
+	}
+
+	Bitmap generateBitmap(byte[] byteArray, int width, int height) {
+		int[] pixels = new int[width * height];
+		for (int row = 0; row < height; row++) {
+			for (int col = 0; col < width; col++) {
+				int position = row * width + col;
+				int r = byteArray[position * 4] & 0xff;
+				int g = byteArray[position * 4 + 1] & 0xff;
+				int b = byteArray[position * 4 + 2] & 0xff;
+				// flip the image vertically
+				pixels[(height - 1 - row) * width + col] = Color.rgb(r, g, b);
+			}
+		}
+		return Bitmap.createBitmap(pixels, width, height,
+				Bitmap.Config.ARGB_8888);
+	}
+
+	void returnScreenshotToUI(final GVRScreenshotCallback callback,
+			ByteBuffer readbackBuffer, int width, int height) {
+		final Bitmap capturedBitmap = generateBitmap(readbackBuffer.array(),
+				width, height);
+		mActivity.runOnUiThread(new Runnable() {
+			public void run() {
+				callback.onScreenCaptured(capturedBitmap);
+			}
+		});
+	}
+
+	void onDrawEyeView(int eye, float fovDegrees) {
+		mCurrentEye = eye;
+		if (!(mSensoredScene == null || !mMainScene.equals(mSensoredScene))) {
+			GVRCameraRig mainCameraRig = mMainScene.getMainCameraRig();
+
+			// if screenshot is enabled, initialize the readback buffer
+			ByteBuffer readbackBuffer = null;
+			int width = 0, height = 0;
+			if (mScreenshotCenterEnabled || mScreenshotLeftEnabled
+					|| mScreenshotRightEnabled) {
+				width = mRenderBundle.getRightRenderTexture().getWidth();
+				height = mRenderBundle.getRightRenderTexture().getHeight();
+				readbackBuffer = ByteBuffer.allocateDirect(width * height * 4);
+				readbackBuffer.order(ByteOrder.nativeOrder());
+			}
+
+			// if mScreenshotCenterEnabled is true, capture center eye
+			if (mScreenshotCenterEnabled) {
+				mainCameraRig.predict(4.0f / 60.0f);
```

Comment:
Shouldn't the `predict()` value be halfway between the left and right eye values?